templates/shortcodes/mermaid.html

```diff
@@ -0,0 +1,22 @@
+{% if body %}  {# inline diagram #}
+    {% set data = body %}
+{% else %}     {# load from file #}
+    {% if section %}
+        {% set base_path="content" ~ section.path %}
+    {% else %}
+        {% set base_path="content" ~ page.path %}
+    {% endif %}
+    {% set data = load_data(path=base_path ~ path) %}
+{% endif%}
+<div class="mermaid">
+{{ data }}
+</div>
+
+<!-- Use latest mermaid renderer from the online CDN by default, -->
+<script src="https://cdn.jsdelivr.net/npm/mermaid/dist/mermaid.min.js"></script>
+<!-- but fall back to a local version if the CDN load failed (e.g. if using docs offline) -->
+<script window.mermaid || document.write('<script src="/js/mermaid.min.js">\x3C/script>')></script>
+
+<!-- when this javascript is loaded and initialized it will look for divs with the class mermaid.
+     it will replace the content of those divs with the generated svg -->
+<script>mermaid.initialize({startOnLoad:true});</script>
```

Comment:
Doesn't it make more sense to put it under `header.html` or `base.html` ?
With that the script will be included only once when the shortcode is used and not every time.

Reply:
I considered that, but then it's also included on every page without a diagram, right? I'm assuming that most pages won't have diagrams, and that pages that do have diagrams will generally only have one (or at least a small number).